apps/api/src/domain/models/__tests__/query-helpers/assert-query-response.ts

```diff
@@ -0,0 +1,29 @@
+import { INestApplication } from '@nestjs/common';
+import * as request from 'supertest';
+import { HttpStatusCode } from '../../../../app/constants/httpStatusCodes';
+
+interface AssertQueryResultParams<TResponseBody = unknown> {
+    app: INestApplication;
+    seedInitialState: () => Promise<void>;
+    endpoint: string;
+    expectedStatus: HttpStatusCode;
+    checkResponseBody?: (body: TResponseBody) => Promise<void>;
+}
+
+export const assertQueryResult = async ({
```

Comment:
I wonder if this name should have something to do with HTTP Status?  If I understand what it's doing, it's not going to be checking for number of results or anything to do with the response body?

Reply:
Oh, now I see `checkResponseBody` at the bottom.  Awesome!!

Reply:
Haha. Glad you sorted that out!

Reply:
But the assertion on the body is optional if all you want to do is assert the status. You probably saw one of those use cases first.